### PR TITLE
Bug 1837961 - Use kotlin coroutines flow extensions 

### PR DESCRIPTION
**Pull request by rahulsainani**
### What
`ifChanged` and `ifChanged(transform: (T) -> R)` are same in behaviour to existing kotlin flow extensions `disctinctUntilChanged` and `distinctUntilChangedBy` in `kotlin.coroutines.flow` package. This PR replaces them.

### How
This is done in 2 parts.
1. Replace the internal implementation of `ifChanged` and make sure all the tests are green, even on CI. ✅ 
2. Migrate the usages of `ifChanged` to `distinctUntilChanged` and `distinctUntilChangedBy` and remove `ifChanged` functions. ✅ 

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1837961

### CODE REVIEW

File: focus-android/app/src/main/java/org/mozilla/focus/input/InputToolbarIntegration.kt
Comment (rahulsainani) [inline focus-android/app/src/main/java/org/mozilla/focus/input/InputToolbarIntegration.kt:135]: Only an Indentation change, couldn't avoid it because of detekt's line length issue, unfortunately

File: android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlinx/coroutines/flow/Flow.kt
Comment (t-p-white) [inline android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlinx/coroutines/flow/Flow.kt:12]: I like the examples on these funcs, shame the Kotlin docs don't do the same

Comment (rahulsainani) [reply android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlinx/coroutines/flow/Flow.kt:12]: Checkout https://flowmarbles.com/#distinctUntilChanged

**Review by t-p-white — COMMENTED** [review]
Great observation to update this :🙌 

lgtm, but as this is a lib change I'm going to hold off on approval as I think it'd beneficial to get another set of eyes on it

**Comment by rahulsainani** [comment]
> Great observation to update this 🙌
> 
> lgtm, but as this is a lib change I'm going to hold off on approval as I think it'd beneficial to get another set of eyes on it

Alright, thanks for taking the time to review 🙌

**Review by t-p-white — APPROVED** [review]
Re-reviewed this in more depth

**Review by MatthewTighe — APPROVED** [review]
🚗 I like the names less tbh but am otherwise fully in favor of using language tooling over custom tooling. Great find!

**Pull request merged**
